R/module_nested_tabs.R

```diff
@@ -214,7 +214,11 @@ srv_teal_module.teal_modules <- function(id,
           datasets = datasets,
           slices_global = slices_global,
           reporter = reporter,
-          is_active = reactive(is_active() && input$active_tab == module_id)
+          is_active = reactive(
+            is_active() &&
+              input$active_tab == module_id &&
+              identical(data_load_status(), "ok")
```

Comment:
Theoretically `input$active_tab` can't be `module_id` as tabs are disabled when data is not "ok". Is this extra condition only to extra-security?

Reply:
Extra performance ensuring that reactivity is only called after we get data.

We can remove this. I added this because I tried to fix an unrelated bug (see line below).

🔴 **HOWEVER**, I'm seeing an bug that modifying filters does not trigger reactivity in modules.

I'm trying to solve this, it seems that an observe event downstream is not triggering the reactivity chain.

[Screencast from 2024-10-10 08-46-59.webm](https://github.com/user-attachments/assets/4b9df19b-af06-4051-afd7-09fb00aa262a)

<details>

<summary>Simple app</summary>

```r
devtools::load_all("teal")

my_transformers <- list(
  teal_transform_module(
    label = "Keep first 6 from IRIS",
    ui = function(id) {
      ns <- NS(id)
      div(
        checkboxInput(ns("check"), label = "Toggle `head(iris)`"),
      )
    },
    server = function(id, data) {
      moduleServer(id, function(input, output, session) {
        eventReactive(input$check, {
          print("Check triggered")
          req(data())
          if (input$check) {
            within(data(), iris <- head(iris, 6))
          } else {
            data()
          }
        })
      })
    }
  )
)

data <- teal_data(iris = iris)
app <- init(
  data = data,
  modules = modules(
    example_module("Module with transformations 1", transformers = my_transformers),
    example_module("Module 2 (no transf.)")
  ),
  filter = teal::teal_slices(
    module_specific = TRUE,
    teal.slice::teal_slice("iris", "Species", selected = c("setosa", "versicolor"))
  ),
  title = "yada"
)

shinyApp(app$ui, app$server)
```

</details>

Reply:
note: this bug is on `main` as well